core/dbt/graph/selector_methods.py

```diff
@@ -320,7 +326,7 @@ def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[Uniqu
                 ymlfp = Path(pfp)
                 if ymlfp in paths:
                     yield node
-            elif any(parent in paths for parent in ofp.parents):
+            if any(parent in paths for parent in ofp.parents):
```

Comment:
this `elif` was not being hit if a node had a `patch_path`. Given these 3 conditions are all independent predicates for path selection matching, separate if statements seemed appropriate to me.